app/assets/javascripts/components/question_tracker/image_question.js.coffee

```diff
@@ -0,0 +1,88 @@
+{div, label, input, textarea, img, option, select} = React.DOM
+
+modulejs.define 'components/question_tracker/image_question',
+  ['components/question_tracker/throttle_mixin', 'components/itsi_authoring/rich_text_editor'],
+  ( ThrottleMixin, RichTextEditorClass) ->
+
+    RichTextEditor = React.createFactory RichTextEditorClass
+
+    React.createClass
+      mixins:
+        [ThrottleMixin]
+
+      getInitialState: ->
+        @resetState()
+
+      sourceOptions: ->
+        [
+          {value: 'Shutterbug', label: 'Snap Shot'},
+          {value: 'Drawing', label: 'Drawing'}
+        ]
+
+      resetState: ->
+        prompt: @props.question.prompt
+        bg_url: @props.question.bg_url
+        drawing_prompt: @props.question.drawing_prompt
+        bg_source: @props.question.bg_source
+
+      cancel: ->
+        @setState @resetState()
+
+      updatePrompt: (v) ->
+        @setState
+          prompt:v
+        , => @throttle(200, @save)
+
+      updateDrawingPrompt: (v) ->
+        @setState
+          drawing_prompt:v
+          , => @throttle(200, @save)
+
+      updateBgUrl: (evt) ->
+        @setState
+          bg_url:evt.target.value
+        , @save
+
```

Comment:
The text input listen to on-blur.  The prompts listen to keyup events, which fire all the time.